src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -0,0 +1,522 @@
+    }
+
+    /**
+     * Returns string value.
+     * @return string value
+     */
+    @Override
+    public String getStringValue() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns CharSequence string value.
+     * @return CharSequence string value
+     */
+    // -@cs[AbbreviationAsWordInName] Implements method from the interface.
+    // No way to change method name, because interface is the part of the Saxon library.
+    @Override
+    public CharSequence getStringValueCS() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns fingerprint.
+     * @return fingerprint
+     */
+    @Override
+    public int getFingerprint() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns localpart.
+     * @return localpart
+     */
+    @Override
+    public String getLocalPart() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns display name.
+     * @return display name
+     */
+    @Override
+    public String getDisplayName() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns prefix.
+     * @return prefix
+     */
+    @Override
+    public String getPrefix() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns type of the schema.
+     * @return type of the schema
+     */
+    @Override
+    public SchemaType getSchemaType() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns AtomicSequence.
+     * @return AtomicSequence
+     * @throws XPathException throws exception
+     */
+    @Override
+    public AtomicSequence atomize() throws XPathException {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns attribute value.
+     * @param namespace namespace
+     * @param localPart actual name of the attribute
+     * @return attribute value
+     */
+    @Override
+    public String getAttributeValue(String namespace, String localPart) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Generate id method.
+     * @param fastStringBuffer fastStringBuffer
+     */
+    @Override
+    public void generateId(FastStringBuffer fastStringBuffer) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Copy method.
+     * @param receiver receiver
+     * @param index index
+     * @param location location
+     * @throws XPathException throws exception
+     */
+    @Override
+    public void copy(Receiver receiver, int index, Location location) throws XPathException {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * To be implemented.
+     * @return UnsupportedOperationException exception
+     */
+    protected UnsupportedOperationException throwUnsupportedOperationException() {
+        return new UnsupportedOperationException("Not implemented yet");
+    }
```

Comment:
Abstract class should give all children common behavior, but in this case it is not. It gives simplification in amounts of methods to override but increase complexity of logic as engineer have to know all nuances of what is overridden and what is not. All implementations of NodeInfo should clearly define what they do on each method.
"copy-paste" is better than dependency between two classes, as it cost you nothing in support. If some new method appear in interface you clearly know how each implementation should support it.
If some big group of methods are never be supported by us, we could make interface with implementations(java8) `UnsupportedInpementationNodeInfo` that have/implement common list of methods with throw expeption and `class ClassNode implements NodeInfo, UnsupportedInpementationNodeInfo`.
If list of common method is small - it just another sign that all classes should have to implement them separately.

@rnveach , please share your ideas on this.